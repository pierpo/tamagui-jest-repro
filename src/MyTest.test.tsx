import { render } from "@testing-library/react-native";

import Home from "../app/index";
import { TamaguiProvider } from "tamagui";
import config from "../tamagui.config";

const Provider = ({children}) => <TamaguiProvider config={config}>{children}</TamaguiProvider>

describe("Home.screen.tsx", () => {
  it("renders correctly", () => {
    const component = render(<Provider><Home /></Provider>);
    expect(component).toMatchSnapshot();
  });
});
