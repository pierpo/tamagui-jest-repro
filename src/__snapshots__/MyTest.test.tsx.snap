// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.screen.tsx renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "#050505",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "paddingBottom": 18,
      "paddingLeft": 18,
      "paddingRight": 18,
      "paddingTop": 18,
    }
  }
>
  <View
    style={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
        "maxWidth": 600,
      }
    }
  >
    <Text
      accessibilityRole="header"
      selectable={true}
      style={
        {
          "color": "#fff",
          "display": "flex",
          "fontFamily": "Inter",
          "fontSize": 46,
          "fontWeight": "300",
          "letterSpacing": 0,
          "lineHeight": 56,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "textAlign": "center",
        }
      }
      suppressHighlighting={true}
    >
      Welcome to Tamagui.
    </Text>
    <View
      pointerEvents="none"
      style={
        {
          "alignItems": "stretch",
          "height": 18,
          "minHeight": 18,
          "minWidth": 18,
          "width": 18,
        }
      }
    />
    <Text
      selectable={true}
      style={
        {
          "color": "#fff",
          "display": "flex",
          "fontFamily": "Inter",
          "fontSize": 14,
          "fontWeight": "300",
          "letterSpacing": 0,
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      suppressHighlighting={true}
    >
      Here's a basic starter to show navigating from one screen to another.
    </Text>
  </View>
  <View
    pointerEvents="none"
    style={
      {
        "alignItems": "stretch",
        "height": 18,
        "minHeight": 18,
        "minWidth": 18,
        "width": 18,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityRole="button"
      cursor="pointer"
      focusable={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#232323",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 9,
          "borderBottomRightRadius": 9,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 9,
          "borderTopRightRadius": 9,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": 44,
          "justifyContent": "center",
          "paddingLeft": 18,
          "paddingRight": 18,
        }
      }
    >
      <Text
        cursor="pointer"
        lineBreakMode="clip"
        numberOfLines={1}
        style={
          {
            "color": "#fff",
            "display": "flex",
            "flexGrow": 0,
            "flexShrink": 1,
            "fontFamily": "Inter",
            "fontSize": 14,
            "fontWeight": "300",
            "letterSpacing": 0,
            "lineHeight": 24,
          }
        }
        suppressHighlighting={true}
        userSelect="none"
      >
        Go to user page
      </Text>
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "alignItems": "stretch",
          "height": 10,
          "minHeight": 10,
          "minWidth": 10,
          "width": 10,
        }
      }
    />
    <View
      accessibilityRole="button"
      cursor="pointer"
      focusable={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#232323",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 9,
          "borderBottomRightRadius": 9,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 9,
          "borderTopRightRadius": 9,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": 44,
          "justifyContent": "center",
          "paddingLeft": 18,
          "paddingRight": 18,
        }
      }
    >
      <Text
        cursor="pointer"
        lineBreakMode="clip"
        numberOfLines={1}
        style={
          {
            "color": "#fff",
            "display": "flex",
            "flexGrow": 0,
            "flexShrink": 1,
            "fontFamily": "Inter",
            "fontSize": 14,
            "fontWeight": "300",
            "letterSpacing": 0,
            "lineHeight": 24,
          }
        }
        suppressHighlighting={true}
        userSelect="none"
      >
        Go to tabs page
      </Text>
    </View>
  </View>
  <View
    pointerEvents="none"
    style={
      {
        "alignItems": "stretch",
        "height": 18,
        "minHeight": 18,
        "minWidth": 18,
        "width": 18,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "stretch",
          "borderBottomColor": "hsl(154, 49.7%, 14.9%)",
          "borderBottomLeftRadius": 9,
          "borderBottomRightRadius": 9,
          "borderBottomWidth": 1,
          "borderLeftColor": "hsl(154, 49.7%, 14.9%)",
          "borderLeftWidth": 1,
          "borderRightColor": "hsl(154, 49.7%, 14.9%)",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "hsl(154, 49.7%, 14.9%)",
          "borderTopLeftRadius": 9,
          "borderTopRightRadius": 9,
          "borderTopWidth": 1,
          "flexDirection": "column",
          "transform": [
            {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        accessibilityRole="link"
        cursor="pointer"
        href="https://twitter.com/natebirdman"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "hsl(155, 44.2%, 8.4%)",
            "borderBottomColor": "hsl(154, 49.7%, 14.9%)",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
            "borderLeftColor": "hsl(154, 49.7%, 14.9%)",
            "borderRightColor": "hsl(154, 49.7%, 14.9%)",
            "borderTopColor": "hsl(154, 49.7%, 14.9%)",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "disabled": undefined,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "justifyContent": "space-between",
            "maxWidth": "100%",
            "minHeight": 44,
            "overflow": "hidden",
            "paddingBottom": 10,
            "paddingLeft": 18,
            "paddingRight": 18,
            "paddingTop": 10,
            "width": "100%",
          }
        }
        target="_blank"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="14"
          bbWidth="14"
          fill="none"
          focusable={false}
          height={14}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#fff"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 14,
                "width": 14,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={14}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          >
            <RNSVGPath
              d="M22 4s-.7 2.1-2 3.4c1.6 10-9.4 17.3-18 11.6 2.2.1 4.4-.6 6-2C3 15.5.5 9.6 3 5c2.2 2.6 5.6 4.1 9 4-.9-4.2 4-6.6 7-3.8 1.1 0 3-1.2 3-1.2z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                ]
              }
              stroke={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          pointerEvents="none"
          style={
            {
              "alignItems": "stretch",
              "height": 14,
              "minHeight": 14,
              "minWidth": 14,
              "width": 14,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "stretch",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            cursor="default"
            lineBreakMode="clip"
            numberOfLines={1}
            style={
              {
                "color": "#fff",
                "display": "flex",
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Inter",
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0,
                "lineHeight": 24,
              }
            }
            suppressHighlighting={true}
          >
            Nate
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "stretch",
            "borderBottomColor": "hsl(154, 49.7%, 14.9%)",
            "borderBottomWidth": 1,
            "borderLeftColor": "hsl(154, 49.7%, 14.9%)",
            "borderLeftWidth": 0,
            "borderRightColor": "hsl(154, 49.7%, 14.9%)",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "hsl(154, 49.7%, 14.9%)",
            "borderTopWidth": 0,
            "flex": 1,
            "flexShrink": 0,
            "height": 0,
            "maxHeight": 0,
            "transform": [
              {
                "translateY": -0.5,
              },
            ],
          }
        }
      />
      <View
        accessibilityRole="link"
        cursor="pointer"
        href="https://github.com/tamagui/tamagui"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "hsl(155, 44.2%, 8.4%)",
            "borderBottomColor": "hsl(154, 49.7%, 14.9%)",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
            "borderLeftColor": "hsl(154, 49.7%, 14.9%)",
            "borderRightColor": "hsl(154, 49.7%, 14.9%)",
            "borderTopColor": "hsl(154, 49.7%, 14.9%)",
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "disabled": undefined,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "justifyContent": "space-between",
            "maxWidth": "100%",
            "minHeight": 44,
            "overflow": "hidden",
            "paddingBottom": 10,
            "paddingLeft": 18,
            "paddingRight": 18,
            "paddingTop": 10,
            "width": "100%",
          }
        }
        target="_blank"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="14"
          bbWidth="14"
          fill="none"
          focusable={false}
          height={14}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#fff"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 14,
                "width": 14,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={14}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          >
            <RNSVGPath
              d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                ]
              }
              stroke={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
            />
            <RNSVGPath
              d="M9 18c-4.51 2-5-2-7-2"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                ]
              }
              stroke={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          pointerEvents="none"
          style={
            {
              "alignItems": "stretch",
              "height": 14,
              "minHeight": 14,
              "minWidth": 14,
              "width": 14,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "stretch",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            cursor="default"
            lineBreakMode="clip"
            numberOfLines={1}
            style={
              {
                "color": "#fff",
                "display": "flex",
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Inter",
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0,
                "lineHeight": 24,
              }
            }
            suppressHighlighting={true}
          >
            Tamagui
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "stretch",
            "borderBottomColor": "hsl(154, 49.7%, 14.9%)",
            "borderBottomWidth": 1,
            "borderLeftColor": "hsl(154, 49.7%, 14.9%)",
            "borderLeftWidth": 0,
            "borderRightColor": "hsl(154, 49.7%, 14.9%)",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "hsl(154, 49.7%, 14.9%)",
            "borderTopWidth": 0,
            "flex": 1,
            "flexShrink": 0,
            "height": 0,
            "maxHeight": 0,
            "transform": [
              {
                "translateY": -0.5,
              },
            ],
          }
        }
      />
      <View
        accessibilityRole="link"
        cursor="pointer"
        href="https://github.com/ivopr/tamagui-expo"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "hsl(155, 44.2%, 8.4%)",
            "borderBottomColor": "hsl(154, 49.7%, 14.9%)",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderLeftColor": "hsl(154, 49.7%, 14.9%)",
            "borderRightColor": "hsl(154, 49.7%, 14.9%)",
            "borderTopColor": "hsl(154, 49.7%, 14.9%)",
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "disabled": undefined,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "justifyContent": "space-between",
            "maxWidth": "100%",
            "minHeight": 44,
            "overflow": "hidden",
            "paddingBottom": 10,
            "paddingLeft": 18,
            "paddingRight": 18,
            "paddingTop": 10,
            "width": "100%",
          }
        }
        target="_blank"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="14"
          bbWidth="14"
          fill="none"
          focusable={false}
          height={14}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#fff"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 14,
                "width": 14,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={14}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          >
            <RNSVGPath
              d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                ]
              }
              stroke={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
            />
            <RNSVGPath
              d="M9 18c-4.51 2-5-2-7-2"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                ]
              }
              stroke={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          pointerEvents="none"
          style={
            {
              "alignItems": "stretch",
              "height": 14,
              "minHeight": 14,
              "minWidth": 14,
              "width": 14,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "stretch",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            cursor="default"
            lineBreakMode="clip"
            numberOfLines={1}
            style={
              {
                "color": "#fff",
                "display": "flex",
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Inter",
                "fontSize": 14,
                "fontWeight": "300",
                "letterSpacing": 0,
                "lineHeight": 24,
              }
            }
            suppressHighlighting={true}
          >
            This Template
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
